MeshLib/MeshSearch/ElementSearch.cpp

```diff
@@ -42,30 +42,33 @@ std::size_t ElementSearch::searchByMaterialID(int const matID)
 	return matchedIDs.size();
 }
 
-std::size_t ElementSearch::searchByElementType(MeshElemType eleType)
+template <typename Container, typename Predicate>
+std::vector<std::size_t> filter(Container const& container, Predicate const& p)
```

Comment:
This `filter` is relatively specific to the ElementSearch, so not directly usable by NodeSearch.

But for future (given some time to work on it) I like to the general filter/map/reduce  functions on meshes and lists of elements/nodes.